examples/inspection/plot_permutation_importance.py

```diff
@@ -0,0 +1,178 @@
+# generalize well enough to the test set thanks to the built-in bagging of
+# random forests.
+#
+# It might be possible to trade some accuracy on the training set for a
+# slightly better accuracy on the test set by limiting the capacity of the
+# trees (for instance by setting ``min_samples_leaf=5`` or
+# ``min_samples_leaf=10``) so as to limit overfitting while not introducing too
+# much underfitting.
+#
+# However let's keep our high capacity random forest model for now so as to
+# illustrate some pitfalls with feature importance on variables with many
+# unique values.
+print("RF train accuracy: %0.3f" % rf.score(X_train, y_train))
+print("RF test accuracy: %0.3f" % rf.score(X_test, y_test))
+
+
+##############################################################################
+# Tree's Feature Importance from Mean Decrease in Impurity (MDI)
+# --------------------------------------------------------------
+# The tree based feature importance ranks the numerical features to be the
+# most important features. As a result, the non-predictive ``random_num``
+# variable is ranked the most important!
+#
+# This problem stems from two limitations of RF feature importances:
+#
+# - RF importances are biased towards high cardinality features;
+# - RF importances are computed on training set statistics and therefore do not
+#   reflect the ability of feature to be useful to make predictions that
+#   generalize to the test set (when the model has enough capacity).
+ohe = (rf.named_steps['preprocess']
+         .named_transformers_['cat']
+         .named_steps['onehot'])
+feature_names = []
+for col, cats in zip(categorical_columns, ohe.categories_):
+    for cat in cats:
+        feature_names.append("{}_{}".format(col, cat))
+feature_names = np.array(feature_names + numerical_columns)
+
+tree_feature_importances = (
+    rf.named_steps['classifier'].feature_importances_)
+sorted_idx = tree_feature_importances.argsort()
+
+y_ticks = np.arange(0, len(feature_names))
+_, ax = plt.subplots(figsize=(10, 8))
+ax.barh(y_ticks, tree_feature_importances[sorted_idx])
+ax.set_yticklabels(feature_names[sorted_idx])
+ax.set_yticks(y_ticks)
+ax.set_title("Random Forest Feature Importances (MDI)")
+
+
+##############################################################################
+# As an alternative, the permutation importances of ``rf`` are computed on a
+# held out test set. This shows that the low cardinality categorical feature,
+# ``sex`` is the most important feature.
+#
+# Also note that both random features have very low importances (close to 0) as
+# expected.
+result = permutation_importance(rf, X_test, y_test, n_repeats=10,
+                                random_state=42)
+sorted_idx = result.importances_mean.argsort()
+
+# sphinx_gallery_thumbnail_number = 2
+_, ax = plt.subplots()
+ax.boxplot(result.importances[sorted_idx].T,
+           vert=False, labels=X_test.columns[sorted_idx])
```

Comment:
Some of the features names used as tick labels in this boxplot and the following are cropped in the rendered png file:

https://63335-843222-gh.circle-artifacts.com/0/doc/auto_examples/inspection/plot_permutation_importance.html#tree-s-feature-importance-from-mean-decrease-in-impurity-mdi

I am not sure how to ensure that matplotlib use the good layout to avoid this issue.

Reply:
Should be fixed now. (We needed to call `fig.tight_layout()`)